File: Acupay/Core Forms/PayStub/Entity/Position.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports AccuPay.JobLevels

Namespace Global.AccuPay.Entity

    <Table("position")>
    Public Class Position

        <Key>
        Public Property RowID As Integer?

        Public Property Name As String

        Public Property Created As Date

        Public Property CreatedBy As Integer?

        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        Public Property ParentPositionID As Integer?

        Public Property DivisionID As Integer?

        Public Property JobLevelID As Integer?

        Public Property LevelNumber As Integer

        <ForeignKey("JobLevelID")>
        Public Overridable Property JobLevel As JobLevel

    End Class

End Namespace

File: AccuPay.Core/Entities/Position.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Core.Entities
{
    [Table("position")]
    public class Position : OrganizationalEntity
    {
        public int? ParentPositionID { get; set; }

        public int? DivisionID { get; set; }

        public int? JobLevelID { get; set; }

        [Column("PositionName")]
        public string Name { get; set; }

        public int LevelNumber { get; set; }

        [ForeignKey("DivisionID")]
        public virtual Division Division { get; set; }

        [ForeignKey("JobLevelID")]
        public virtual JobLevel JobLevel { get; set; }
    }
}
